pkg/controller/installation/products/rhssouser/reconciler.go

```diff
@@ -0,0 +1,362 @@
+	kc := &aerogearv1.Keycloak{}
+	// if this errors, it can be ignored
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakName, Namespace: r.Config.GetNamespace()}, kc)
+	if err == nil && string(r.Config.GetProductVersion()) != kc.Status.Version {
+		r.Config.SetProductVersion(kc.Status.Version)
+		r.ConfigManager.WriteConfig(r.Config)
+	}
+
+	r.logger.Info("checking ready status for user-sso")
+	kcr := &aerogearv1.KeycloakRealm{}
+
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakRealmName, Namespace: r.Config.GetNamespace()}, kcr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get keycloak realm custom resource")
+	}
+
+	if kcr.Status.Phase == aerogearv1.PhaseReconcile {
+		err = r.exportConfig(ctx, serverClient)
+		if err != nil {
+			return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to write user-sso config")
+		}
+
+		err = r.setupOpenshiftIDP(ctx, inst, kcr, serverClient)
+		if err != nil {
+			return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to setup Openshift IDP for user-sso")
+		}
+
+		logrus.Infof("Keycloak has successfully processed the keycloakRealm for user-sso")
+		return v1alpha1.PhaseCompleted, nil
+	}
+
+	r.logger.Infof("user-sso KeycloakRealm status phase is: %s", kcr.Status.Phase)
+	return v1alpha1.PhaseInProgress, nil
+}
+
+func (r *Reconciler) exportConfig(ctx context.Context, serverClient pkgclient.Client) error {
+	kc := &aerogearv1.Keycloak{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      keycloakName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: keycloakName, Namespace: r.Config.GetNamespace()}, kc)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not retrieve keycloak custom resource for keycloak config for user-sso")
+	}
+	kcAdminCredSecretName := kc.Spec.AdminCredentials
+
+	kcAdminCredSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kcAdminCredSecretName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: kcAdminCredSecretName, Namespace: r.Config.GetNamespace()}, kcAdminCredSecret)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not retrieve keycloak admin credential secret for keycloak config for user-sso")
+	}
+	kcURLBytes := kcAdminCredSecret.Data["SSO_ADMIN_URL"]
+	r.Config.SetRealm(keycloakRealmName)
+	r.Config.SetHost(string(kcURLBytes))
+	err = r.ConfigManager.WriteConfig(r.Config)
+	if err != nil {
+		return pkgerr.Wrap(err, "could not update keycloak config for user-sso")
+	}
+	return nil
+}
+
+func (r *Reconciler) setupOpenshiftIDP(ctx context.Context, inst *v1alpha1.Installation, kcr *aerogearv1.KeycloakRealm, serverClient pkgclient.Client) error {
+	oauthc := &oauthv1.OAuthClient{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: oauthId,
+		},
+		Secret: clientSecret,
+		RedirectURIs: []string{
+			r.Config.GetHost() + "/auth/realms/openshift/broker/openshift-v4/endpoint",
```

Comment:
Should `openshift` be changed to the new realm name?

Reply:
:+1: redirect uri is pointing to https://sso-integreatly-user-sso.apps.aucunnin.cloudservices.rhmw.io/auth/realms/user-sso/broker/openshift-v4/endpoint